### PR TITLE
Fix setting original path of filesystem object to not have extra trailing slash

### LINKED ISSUE #10956: Get-Item with a drive's root dir stringifies incorrectly
This is a regression from Windows PowerShell.

When accessing a drive's root directory, an extra path separator is appended on stringification.

# Steps to reproduce

```powershell
$root = ('/', 'C:\')[$env:OS -eq 'Windows_NT']
Push-Location ('/', 'C:\')[$env:OS -eq 'Windows_NT']

(Get-Item /).ToString() | Should -Be $root
(Get-Item .).ToString() | Should -Be $root

Pop-Location 
```

# Expected behavior

The tests should pass.

# Actual behavior

The tests fails; e.g., on Windows:

```none
Expected: 'C:\'
But was:  'C:\\'
```

# Environment data

```none
PowerShell Core 7.0.0-preview.5
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

In the method to build the correct casing of a path, it handled the root of the drive, but later in the loop since the original path had a trailing slash, it added another one.  Fix is to detect if the resulting path already has a trailing slash and don't add it.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/10956

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by SteveL-MSFT** [comment]
Codacy uses PSSA which misidentified ternary operator as cmdlet.  I opened https://github.com/PowerShell/PSScriptAnalyzer/issues/1362

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -159,7 +159,7 @@ private static string GetCorrectCasedPath(string path)
                     }
                 }
 
-                if (path.EndsWith(StringLiterals.DefaultPathSeparator))
+                if (path.EndsWith(StringLiterals.DefaultPathSeparator) && !exactPath.EndsWith(StringLiterals.DefaultPathSeparator))
                 {
                     return exactPath + StringLiterals.DefaultPathSeparator;
                 }
```
Comment (iSazonov) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: It seems we could remove the lines at all.

Comment (SteveL-MSFT) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: `path` is the original path that may have incorrect casing, so if it originally contained a trailing slash, we want to add it back.  In the case of the root of the drive, it already has a trailing slash so we don't want to add another one.

Comment (iSazonov) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: > // This handles the trailing slash case

There is already a branch for this. So proposal is:
```c#
                foreach (string item in path.Split(StringLiterals.DefaultPathSeparator))
                {
                    if (itemsToSkip-- > 0)
                    {
                        // This handles the UNC server and share and 8.3 short path syntax
                        exactPath += item + StringLiterals.DefaultPathSeparator;
                        continue;
                    }
                    else if (string.IsNullOrEmpty(exactPath))
                    {
                        // This handles the drive letter or / root path start
                        exactPath = item + StringLiterals.DefaultPathSeparator;
                    }
                    else if (string.IsNullOrEmpty(item))
                    {
                        // This handles the trailing slash case
                        if (!exactPath.EndsWith(StringLiterals.DefaultPathSeparator))
                        {
                            exactPath += StringLiterals.DefaultPathSeparator;
                        }

                        break;
                    }
                    else if (item.Contains('~'))
                    {
                        // This handles short path names
                        exactPath += StringLiterals.DefaultPathSeparator + item;
                    }
                    else
                    {
                        exactPath = Directory.GetFileSystemEntries(exactPath, item).First();
                    }
                }

                return exactPath;
```

Comment (SteveL-MSFT) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Ok, got it.

**Review by iSazonov — APPROVED** [review]

**Pull request merged**